library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt

```diff
@@ -172,29 +188,29 @@ class ProtocolClient(
         headers: Headers,
         methodSpec: MethodSpec<Input, Output>,
     ): BidirectionalStreamInterface<Input, Output> {
-        return bidirectionalStream(methodSpec, headers)
+        return newStream(headers, methodSpec)
     }
 
     override suspend fun <Input : Any, Output : Any> serverStream(
         headers: Headers,
         methodSpec: MethodSpec<Input, Output>,
     ): ServerOnlyStreamInterface<Input, Output> {
-        val stream = bidirectionalStream(methodSpec, headers)
+        val stream = newStream(headers, methodSpec)
         return ServerOnlyStream(stream)
     }
 
     override suspend fun <Input : Any, Output : Any> clientStream(
         headers: Headers,
         methodSpec: MethodSpec<Input, Output>,
     ): ClientOnlyStreamInterface<Input, Output> {
-        val stream = stream(headers, methodSpec)
+        val stream = newStream(headers, methodSpec)
         return ClientOnlyStream(stream)
     }
 
-    private suspend fun <Input : Any, Output : Any> bidirectionalStream(
-        methodSpec: MethodSpec<Input, Output>,
+    private fun <Input : Any, Output : Any> newStream(
         headers: Headers,
-    ): BidirectionalStream<Input, Output> = suspendCancellableCoroutine { continuation ->
```

Comment:
More cleanup: the `suspendCancellableCoroutine` usage here didn't really do anything.

It tried to associate cancellation of the stream with the coroutine being cancelled. But this only does anything for the duration of this method body, and only when the coroutine is suspended. 

Since the method always immediately resumes the continuation, this didn't really do anything in practice. So I've removed it for clarity.

Now, it's no longer a suspendable function (so can't be cancelled while executing anyway). Instead, the previous PR (#213, that adds auto-closing block-based RPC invocations to the conformance client) demonstrates the better (IMO) way to go for making sure that the RPCs are appropriately cancelled when a coroutine gets cancelled.